src/mon/OSDMonitor.cc

```diff
@@ -592,6 +592,11 @@ int OSDMonitor::reweight_by_utilization(int oload,
     std::pair<int, float> osd_util;
     osd_util.first = p->first;
     if (by_pg) {
+      if (pgs_by_osd[p->first] == 0) {
```

Comment:
`pgs_by_osd[p->first]` could read out of the bound if the osd id is greater than the largest osd id serving the specified pools. so maybe we should check in this way:
```c++
if (p->first >= (int)pgs_by_osd.size()) {
```